test/spec/screenSharing/screenSharingSrv.spec.js

```diff
@@ -0,0 +1,235 @@
+
+        var sharerData = {
+            uid: sharerId,
+            isTeacher: false
+        };
+        _deps.ScreenSharingSrv.viewOtherUserScreen(sharerData);
+
+        var expectedRootScreenSharing = {
+            guid: {
+                sharerId: sharerId,
+                viewerId: viewerId,
+                status: _deps.ScreenSharingStatusEnum.PENDING_SHARER.enum,
+                guid: screenSharingDataGuid
+            }
+        };
+        expect(_deps.testStorage.db.screenSharing).toEqual(jasmine.objectContaining(expectedRootScreenSharing));
+
+        var expectedStudentAppUsersObject = {};
+        expectedStudentAppUsersObject[viewerId] = {
+            screenSharing: {
+                guid: true
+            }
+        };
+        expectedStudentAppUsersObject[sharerId] = {
+            screenSharing: {
+                guid: true
+            }
+        };
+
+        var currStudentAppUsersObject = _deps.testStorage.db[_deps.ENV.studentAppName].users;
+        expect(currStudentAppUsersObject).toEqual(jasmine.objectContaining(expectedStudentAppUsersObject));
+    });
+
+    it('given user sharing his screen with me when i confirm sharing then sharing status should ' +
+        'be changed to confirmed', function () {
+        var screenSharingDataGuid = '123456789-data-guid';
+        var sharerId = '123456789-sharer-id';
+        var viewerId = _deps.UserProfileService.__currUserId;
+
+        _deps.testStorage.db.screenSharing = {};
+        _deps.testStorage.db.screenSharing[screenSharingDataGuid] = {
+            guid: screenSharingDataGuid,
+            sharerId: sharerId,
+            viewerId: viewerId,
+            status: _deps.ScreenSharingStatusEnum.PENDING_VIEWER.enum
+        };
+
+        _deps.ScreenSharingSrv.confirmSharing(screenSharingDataGuid);
+
+        var expectedRootScreenSharing = {};
+        expectedRootScreenSharing[screenSharingDataGuid] = angular.copy(_deps.testStorage.db.screenSharing[screenSharingDataGuid]);
+        expectedRootScreenSharing[screenSharingDataGuid].status = _deps.ScreenSharingStatusEnum.CONFIRMED.enum;
+
+        expect(_deps.testStorage.db.screenSharing).toEqual(jasmine.objectContaining(expectedRootScreenSharing));
+    });
+
+    it('given user sharing his screen with me when i end screen sharing then sharing status should ' +
+        'be changed to ended', function () {
+        var screenSharingDataGuid = '123456789-data-guid';
+        var sharerId = '123456789-sharer-id';
+        var viewerId = _deps.UserProfileService.__currUserId;
+
+        _deps.testStorage.db.screenSharing = {};
+        _deps.testStorage.db.screenSharing[screenSharingDataGuid] = {
+            guid: screenSharingDataGuid,
+            sharerId: sharerId,
+            viewerId: viewerId,
+            status: _deps.ScreenSharingStatusEnum.PENDING_VIEWER.enum
+        };
+
+        _deps.ScreenSharingSrv.endSharing(screenSharingDataGuid);
+
+        var expectedRootScreenSharing = {};
```

Comment:
The only diff between this test and the previous one is the status
 status: _deps.ScreenSharingStatusEnum.PENDING_VIEWER.enum
and the actio result - endsharing/confirmsharing maybe create a func for state change test ? not mandatory